File: App_Resources/Android/src/main/java/com/faceclaw/app/FaceclawMediaControllerListener.java
package com.faceclaw.app;

public interface FaceclawMediaControllerListener {
    /** All observed session owners, before filtering or choosing the active player. */
    void onSessionAppsChanged(String appsJson);

    void onStateChange(
            String playbackState,
            String packageName,
            String appName,
            String title,
            String artist,
            String album,
            long positionMs,
            long durationMs,
            float playbackSpeed,
            boolean canPlayPause,
            boolean canSkipNext,
            boolean canSkipPrevious,
            boolean accessEnabled,
            String status
    );
}

File: native/kotlin/shared/src/commonMain/kotlin/com/faceclaw/app/callbacks/FaceclawMediaControllerListener.kt
package com.faceclaw.app

interface FaceclawMediaControllerListener {
    /** All observed session owners, before filtering or choosing the active player. */
    fun onSessionAppsChanged(appsJson: String?): Unit

    fun onStateChange(
        playbackState: String?,
        packageName: String?,
        appName: String?,
        title: String?,
        artist: String?,
        album: String?,
        positionMs: Long,
        durationMs: Long,
        playbackSpeed: Float,
        canPlayPause: Boolean,
        canSkipNext: Boolean,
        canSkipPrevious: Boolean,
        accessEnabled: Boolean,
        status: String?,
    ): Unit
}
